pkg/networkserver/grpc_gsns.go

```diff
@@ -706,24 +647,24 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 					logger.WithField("cid", cmd.CID).WithError(err).Warn("Failed to process MAC command")
 				}
 			}
-			if stored.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_1) < 0 {
-				stored.Session = ses
-				stored.PendingSession = nil
-			} else if stored.PendingSession != nil {
-				handleErr = true
-				return nil, nil, errNoRekey
+			if pending {
+				if stored.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_1) < 0 {
+					stored.Session = stored.PendingSession
+					stored.PendingSession = nil
+					stored.EndDeviceIdentifiers.DevAddr = &pld.DevAddr
+				} else if stored.PendingSession != nil {
+					handleErr = true
+					return nil, nil, errNoRekey
+				}
 			}
+			stored.PendingSession = nil
 			paths = append(paths,
 				"pending_session",
```

Comment:
I see it's here, but it's already set before. I'd prefer keeping this close to assignments.